cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -431,23 +742,36 @@ instance IsCardanoEra era => HasTextEnvelope (TxBody era) where
         MaryEra    -> "TxBodyMary"
 
 
-data ByronTxBodyConversionError =
-       ByronTxBodyEmptyTxIns
-     | ByronTxBodyEmptyTxOuts
-     | ByronTxBodyLovelaceOverflow (TxOut ByronEra)
+-- ----------------------------------------------------------------------------
+-- Constructing transaction bodies
+--
+
+data TxBodyError era =
+       TxBodyEmptyTxIns
+     | TxBodyEmptyTxOuts
+     | TxBodyLovelaceOverflow (TxOut era)
      deriving Show
 
-makeByronTransaction :: [TxIn]
-                     -> [TxOut ByronEra]
-                     -> Either ByronTxBodyConversionError
-                               (TxBody ByronEra)
-makeByronTransaction ins outs = do
-    ins'  <- NonEmpty.nonEmpty ins        ?! ByronTxBodyEmptyTxIns
+
```

Comment:
I think we can remove the `forall era` and `:: CardanoEraStyle era` here

Reply:
It's true, but in the other instances of this pattern in the same module we cannot, and it's probably better to do it the same way for them all.